Dockerfile

```
@@ -8,14 +12,44 @@ COPY ./proxy-init ./proxy-init
 ARG TARGETARCH
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=$TARGETARCH go build -o /out/linkerd2-proxy-init -mod=readonly -ldflags "-s -w" -v ./proxy-init
 
-## Package runtime
-FROM --platform=$TARGETPLATFORM alpine:3.16.2
+##
+## Rust
+##
+
+# Compile from target platform to target arch
+FROM --platform=$TARGETPLATFORM docker.io/library/rust:1.62.1 as rust
+WORKDIR /build
+COPY Cargo.toml Cargo.lock .
+COPY validator /build/
+ARG TARGETARCH
+RUN --mount=type=cache,target=target \
+   --mount=type=cache,from=rust:1.62.1,source=/usr/local/cargo,target=/usr/local/cargo \
+   cargo fetch
+RUN --mount=type=cache,target=target \
+   --mount=type=cache,from=rust:1.62.1,source=/usr/local/cargo,target=/usr/local/cargo \
```

rust was updated in https://github.com/linkerd/linkerd2-proxy-init/commit/204f29f906e5d0d603adc24d98bbad4e6ef4534d
```suggestion
FROM --platform=$TARGETPLATFORM docker.io/library/rust:1.63.0 as rust
WORKDIR /build
COPY Cargo.toml Cargo.lock .
COPY validator /build/
ARG TARGETARCH
RUN --mount=type=cache,target=target \
   --mount=type=cache,from=docker.io/library/rust:1.63.0,source=/usr/local/cargo,target=/usr/local/cargo \
   cargo fetch
RUN --mount=type=cache,target=target \
   --mount=type=cache,from=docker.io/library/rust:1.63.0,source=/usr/local/cargo,target=/usr/local/cargo \
```

Also, we should probably be consistent about using `docker.io/library/` or not